### DISCUSSION TITLE
How to get Wake-up reasons in stm32？

### DISCUSSION THREAD

**Discussion by xtbwqtq** [micropython/micropython#15532] [STM32 / Pyboard]
```machine.wake_reason()```

```Availability: ESP32, WiPy.```
This method does not seem to support stm32

**Comment by bidrohini** [comment]
I think this task will not be easy on STM32 boards. In STM32 microcontrollers, determining the wake-up reason typically involves checking specific registers that record the cause of the wake-up event. This is not as straightforward as using a dedicated method like machine.wake_reason() in platforms like ESP32 or WiPy. To get the wake-up reasons on STM32, you need to check the appropriate power and reset flags using the HAL or LL library functions. For further info, you can post this query to the STM forum. And in case you are planning to make a custom STM32 microcontroller board, you can see here: https://www.pcbway.com/blog/25/Tutorial__How_to_Design_Your_Own_Custom_STM32_Microcontroller_Board.html
